libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2833,12 +2830,32 @@ string YulUtilFunctions::allocationTemporaryMemoryFunction()
 	});
 }
 
-string YulUtilFunctions::releaseTemporaryMemoryFunction()
+string YulUtilFunctions::finalizeAllocationFunction()
+{
+	string functionName = "finalize_allocation";
+	return m_functionCollector.createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(memPtr, size) {
+				let newFreePtr := add(memPtr, <roundUp>(size))
+				// protect against overflow
+				if or(gt(newFreePtr, 0xffffffffffffffff), lt(newFreePtr, memPtr)) { <panic>() }
+				mstore(<freeMemoryPointer>, newFreePtr)
+			}
+		)")
+		("functionName", functionName)
+		("freeMemoryPointer", to_string(CompilerUtils::freeMemoryPointer))
+		("roundUp", roundUpFunction())
+		("panic", panicFunction(PanicCode::ResourceError))
+		.render();
+	});
+}
+
+string YulUtilFunctions::releaseMemoryFunction()
 {
-	string functionName = "releaseTemporaryMemory";
+	string functionName = "release_memory";
 	return m_functionCollector.createFunction(functionName, [&](){
 		return Whiskers(R"(
-			function <functionName>() {
+			function <functionName>(ptr) {
 			}
```

Comment:
Shouldn't technically this have an assertion for `ptr == freeMemoryPointer`?

Reply:
This is unused functionality. We might decide to write a complicated allocator ...

Reply:
I wish there would be a way to have this ensured in a no-overhead version. What is the point having the hypothetical free memory when it is not tested.